### ISSUE TITLE
ValueError 

### ISSUE THREAD

**Issue by gigadeplex** [paulbauriegel/tensorflow-tools#8]
Hi after converting pytorch to onnx (using torch.onnx) then to pb (using onnx-tf) I now want to change format from NCHW to NHWC. Now after running your script I'm getting a ValueError.
```

---------------------------------------------------------------------------

InvalidArgumentError                      Traceback (most recent call last)

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in _create_c_op(graph, node_def, inputs, control_inputs, op_def)
   1653   try:
-> 1654     c_op = pywrap_tf_session.TF_FinishOperation(op_desc)
   1655   except errors.InvalidArgumentError as e:

InvalidArgumentError: Depth of input (322) is not a multiple of input depth of filter (3) for '{{node import/convolution_new_new_2}} = Conv2D[T=DT_FLOAT, data_format="NHWC", dilations=[1, 1, 1, 1], explicit_paddings=[], padding="VALID", strides=[1, 1, 1, 1], use_cudnn_on_gpu=true](import/convolution_new_transp_input_2, import/split)' with input shapes: [1,322,3,322], [3,3,3,64].


During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)

5 frames

<ipython-input-52-b22301f496b5> in <module>()
     38         # Create new Operation
     39         #print(n_org.type, n_org.name, list(n_org.inputs), n_org.node_def.attr['data_format'])
---> 40         op = sess.graph.create_op(op_type=n_org.type, inputs=op_inputs, name=n_org.name+'_new', dtypes=[tf.float32], attrs=atts)
     41         out_tens = sess.graph.get_tensor_by_name(n_org.name+'_new'+':0')
     42         out_trans = tf.transpose(out_tens, [0, 3, 1, 2], name=n_org.name +'_transp_out')

/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/deprecation.py in new_func(*args, **kwargs)
    505                 'in a future version' if date is None else ('after %s' % date),
    506                 instructions)
--> 507       return func(*args, **kwargs)
    508 
    509     doc = _add_deprecated_arg_notice_to_docstring(

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in create_op(***failed resolving arguments***)
   3256         raise TypeError("Input #%d is not a tensor: %s" % (idx, a))
   3257     return self._create_op_internal(op_type, inputs, dtypes, input_types, name,
-> 3258                                     attrs, op_def, compute_device)
   3259 
   3260   def _create_op_internal(

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in _create_op_internal(self, op_type, inputs, dtypes, input_types, name, attrs, op_def, compute_device)
   3325           input_types=input_types,
   3326           original_op=self._default_original_op,
-> 3327           op_def=op_def)
   3328       self._create_op_helper(ret, compute_device=compute_device)
   3329     return ret

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in __init__(self, node_def, g, inputs, output_types, control_inputs, input_types, original_op, op_def)
   1815         op_def = self._graph._get_op_def(node_def.op)
   1816       self._c_op = _create_c_op(self._graph, node_def, inputs,
-> 1817                                 control_input_ops, op_def)
   1818       name = compat.as_str(node_def.name)
   1819     # pylint: enable=protected-access

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in _create_c_op(graph, node_def, inputs, control_inputs, op_def)
   1655   except errors.InvalidArgumentError as e:
   1656     # Convert to ValueError for backwards compatibility.
-> 1657     raise ValueError(str(e))
   1658 
   1659   return c_op

ValueError: Depth of input (322) is not a multiple of input depth of filter (3) for '{{node import/convolution_new_new_2}} = Conv2D[T=DT_FLOAT, data_format="NHWC", dilations=[1, 1, 1, 1], explicit_paddings=[], padding="VALID", strides=[1, 1, 1, 1], use_cudnn_on_gpu=true](import/convolution_new_transp_input_2, import/split)' with input shapes: [1,322,3,322], [3,3,3,64].


```

**Comment by paulbauriegel** [comment]
Ok, either I put a bug in the script with the last changes I made, or I missed a property. So you can try the original script: https://github.com/paulbauriegel/tensorflow-tools/commit/d6b1df965e688d34003546ecfa338ab233245d83#diff-df715ec6a2ebe5af9cdbe20dfe9dcd01 or send give me a link to the model you try to convert, so that I can do some debugging

**Comment by paulbauriegel** [comment]
On thing that already looks strange to me is that the input shapes look like this: `with input shapes: [1,322,3,322], [3,3,3,64].`, the should however look like this: `[1,3,322,322]`. Can you print every nodes in and output shapes? What node lies ahead of the `import/convolution_new_transp_input_2`

**Comment by gigadeplex** [comment]
@paulbauriegel Hi, thank you for responding. Here is a link to the pb file: 
https://drive.google.com/file/d/1-LMWgrb5se7o4jKddLHpF5VsoN5CAzmn/view?usp=sharing
And here is a link to the onnx file:
https://drive.google.com/file/d/1-LOIrcAyY6LjX7V5P5m2lvZEwxhYFRA-/view?usp=sharing

**Comment by gigadeplex** [comment]
@paulbauriegel the original input shape is `1, 3, 320, 320` (from pytorch to onnx)
I must also add that I'm using tensorflow 2.2.0 and so I changed the script to from `tf.x` to `tf.compat.v1.x`

**Comment by paulbauriegel** [comment]
@gigadeplex  Thank you very much, I will look into it tomorrow and notify you once I fixed it.

**Comment by gigadeplex** [comment]
@paulbauriegel Ok thank you for your time!

**Comment by paulbauriegel** [comment]
@gigadeplex It took me a while to understand the problem, but you model is already on the NHWC data_format if that exception comes. So you can directly convert it to TFLite. I made some other observations however I wantet to share. 
1. When I was checking the model you uploaded it, my script worked directly, but because of that https://github.com/onnx/onnx-tensorflow/issues/595 Issue converting it to TFLite didn't work. So I needed to convert you ONNX model first by using `!pip install git+https://github.com/sdmonov/onnx-tensorflow@maxpool_use_pool_op_instead_delation`
2. With the converted model I got the same issue as you. I converted the model without having any GPU installed and when I was looking at the first Conv2D Node I it had that: `attr {  key: "data_format", value {s: "NHWC"}}`
3. Since onnx_tf converted the model directly to NHWC without a GPU installed I thought using it like `prepare(onnx_model, device='CPU')` would do the job, but it didn't. However disabling it with `os.environ["CUDA_VISIBLE_DEVICES"]="-1"` worked perfectly and the model was in NHWC directly.

To summarize:
```
import os
os.environ["CUDA_VISIBLE_DEVICES"]="-1"
import onnx
from onnx_tf.backend import prepare

onnx_model = onnx.load("model.onnx")  # load onnx model
tf_rep = prepare(onnx_model, device='CPU')  # prepare tf representation
tf_rep.export_graph("model2.pb")  # export the model

graph_def_file = "model2.pb"
input_arrays = ["input_1"]
output_arrays = ["output_1"]

converter = tfv1.lite.TFLiteConverter.from_frozen_graph(
        graph_def_file, input_arrays, output_arrays)
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS,
                                       tf.lite.OpsSet.SELECT_TF_OPS]
tflite_model = converter.convert()
open("converted_model.tflite", "wb").write(tflite_model)
```

and you don't even need my script 😀 Hope it helped!
